hail/python/hail/ir/matrix_ir.py

```diff
@@ -229,21 +229,14 @@ def render(self, r):
 
 
 class MatrixAnnotateRowsTable(MatrixIR):
```

Comment:
Maybe outside the scope of this commit: I would find it helpful if there were some documentation on how this class and those like it are consumed, why it exists.

Reply:
do you mean documentation on what MatrixIRs are for, or documentation on the semantics of each IR node and the transformations it applies?

Reply:
This does sound outside the scope of this PR, but that is something we need, for sure. I think in the next few months as everything is solidified in the IR, we can stabilize the interface and document it.

Reply:
Sounds good! Sorry for the delay, have been quite ill.

I think both would be useful, but mostly I meant the latter, because that could have helped me evaluate PR. 

Thanks!